### PR TITLE
Graphics_Engine: Align to Graphics_oM `Vertex` to `RenderPoint` refactoring

### PR DESCRIPTION
**Pull request by alelom**
### NOTE: Depends on 

https://github.com/BHoM/BHoM/pull/1280
   
### Issues addressed by this PR

Closes #2606

### CODE REVIEW

**Review by al-fisher — DISMISSED** [review]
See comment on oM PR https://github.com/BHoM/BHoM/pull/1280#discussion_r697621494 around necessary implementation of extension methods for `IElement0D`

**Review by enarhi — APPROVED** [review]
LGTM!

File: Graphics_Engine/Query/SetGeometry.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            {
+                BH.Engine.Reflection.Compute.RecordError($"Cannot set the geometry of a null {nameof(RenderPoint)}.");
+                return obj;
+            }
+
+            if (point == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError($"Cannot set null geometry to a {nameof(RenderPoint)}.");
+                return obj;
+            }
```
[the remark this answers is not in the document]
Comment (alelom) [reply Graphics_Engine/Query/SetGeometry.cs:48]: Removed. I think I had blindly copied what I have seen done for other objects in the Environment namespace.

File: Graphics_Engine/Query/SetGeometry.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+            var clone = obj.ShallowClone();
+            clone.Point = point.DeepClone();
```
[the remark this answers is not in the document]
Comment (alelom) [reply Graphics_Engine/Query/SetGeometry.cs:51]: Again, I think I have blindly copied another example in the Environment. But I agree, no reason to break the reference with a clone of the Point here.

**Comment by alelom** [comment]
> Some comments below. Leaving as comment rather than request change as not supercritical.

Thanks @IsakNaslundBh , I also noticed I made a more important mistake (`SetGeometry()` was in the `Query` class rather than `Modify`) so I decided to push your suggestion along with that.

**Comment by alelom** [comment]
@BHoMBot check compliance

**Comment by alelom** [comment]
@BHoMBot check compliance

**Comment by alelom** [comment]
@BHoMBot check required

**Comment by pawelbaran** [comment]
@BHoMBot check core

**Comment by alelom** [comment]
@BHoMBot check required

File: Graphics_Engine/Query/Geometry.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Query the geometry that define the input RenderPoint.")]
+        [Input("obj", "RenderPoint to get the geometry of.")]
+        [Output("geom", "Geometry obtained from the RenderPoint.")]
+        public static IGeometry Geometry(this RenderPoint obj)
+        {
+            return obj.Point;
```
Comment (FraserGreenroyd) [inline Graphics_Engine/Query/Geometry.cs:42]: @alelom I think this method is going to be the one causing the `null-handling` failures. It's a new method with no `null` checking.

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by alelom** [comment]
@BHoMBot check required

File: Graphics_Engine/Query/Geometry.cs (hunk 3 above)
Comment (alelom) [reply Graphics_Engine/Query/Geometry.cs:45]: Thanks @FraserGreenroyd

**Comment by alelom** [comment]
@BHoMBot check ready-to-merge

**Review by enarhi — APPROVED** [review]
LGTM, with pending requested changes incorporated

**Review by FraserGreenroyd — APPROVED** [review]
LGTM now

**Comment by FraserGreenroyd** [comment]
@BHoMBot check dataset-compliance
@BHoMBot check copyright-compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot check dataset-compliance
@BHoMBot check copyright-compliance

**Review by FraserGreenroyd — APPROVED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**